pkg/virt-operator/resource/generate/components/daemonsets.go

```diff
@@ -0,0 +1,239 @@
+
+	container.VolumeMounts = []corev1.VolumeMount{}
+
+	container.LivenessProbe = &corev1.Probe{
+		FailureThreshold: 3,
+		Handler: corev1.Handler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Scheme: corev1.URISchemeHTTPS,
+				Port: intstr.IntOrString{
+					Type:   intstr.Int,
+					IntVal: 8443,
+				},
+				Path: "/healthz",
+			},
+		},
+		InitialDelaySeconds: 15,
+		TimeoutSeconds:      10,
+		PeriodSeconds:       45,
+	}
+	container.ReadinessProbe = &corev1.Probe{
+		Handler: corev1.Handler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Scheme: corev1.URISchemeHTTPS,
+				Port: intstr.IntOrString{
+					Type:   intstr.Int,
+					IntVal: 8443,
+				},
+				Path: "/healthz",
+			},
+		},
+		InitialDelaySeconds: 15,
+		TimeoutSeconds:      10,
+		PeriodSeconds:       20,
+	}
+
+	pod.Volumes = []corev1.Volume{}
+
+	type volume struct {
+		name             string
+		path             string
+		mountPath        string
+		mountPropagation *corev1.MountPropagationMode
+	}
+	attachCertificateSecret(pod, VirtHandlerCertSecretName, "/etc/virt-handler/clientcertificates")
+	attachCertificateSecret(pod, VirtHandlerServerCertSecretName, "/etc/virt-handler/servercertificates")
+
+	bidi := corev1.MountPropagationBidirectional
+	// NOTE: the 'kubelet-pods-shortened' volume mounts the same host path as 'kubelet-pods'
+	// This is because that path holds unix domain sockets. Domain sockets fail when they're over
+	// ~ 100 characters, so that shortened volume path is to allow domain socket connections.
+	// It's ridiculous to have to account for that, but that's the situation we're in.
+	volumes := []volume{
+		{"libvirt-runtimes", "/var/run/kubevirt-libvirt-runtimes", "/var/run/kubevirt-libvirt-runtimes", nil},
+		{"virt-share-dir", "/var/run/kubevirt", "/var/run/kubevirt", &bidi},
+		{"virt-lib-dir", "/var/lib/kubevirt", "/var/lib/kubevirt", nil},
+		{"virt-private-dir", "/var/run/kubevirt-private", "/var/run/kubevirt-private", nil},
+		{"device-plugin", "/var/lib/kubelet/device-plugins", "/var/lib/kubelet/device-plugins", nil},
+		{"kubelet-pods-shortened", "/var/lib/kubelet/pods", "/pods", nil},
+		{"kubelet-pods", "/var/lib/kubelet/pods", "/var/lib/kubelet/pods", &bidi},
+		{"node-labeller", "/var/lib/kubevirt-node-labeller", "/var/lib/kubevirt-node-labeller", nil},
+	}
+
+	for _, volume := range volumes {
+		container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
+			Name:             volume.name,
+			MountPath:        volume.mountPath,
+			MountPropagation: volume.mountPropagation,
+		})
+		pod.Volumes = append(pod.Volumes, corev1.Volume{
+			Name: volume.name,
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: volume.path,
+				},
+			},
+		})
+	}
+
+	container.Resources = corev1.ResourceRequirements{
+		Requests: corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("10m"),
+			corev1.ResourceMemory: resource.MustParse("200Mi"),
```

Comment:
I've seen virt-handler consume as much as 183Mi in the wild. I would imagine we don't need to cover all cases, but 200Mi might be cutting it a tiny bit close.

Reply:
I've added ~50Mi padding so 230Mi total. Does that seem reasonable? Or should there be more?